tools/cdi-func-test-proxy/main.go

```diff
@@ -216,19 +213,10 @@ func main() {
 		klog.Fatal(errors.Wrapf(err, "populate certificate directory %s' errored: ", certDir))
 	}
 
-	wg := new(sync.WaitGroup)
+	startServer(httpPort, noBasicAuth, false)
+	startServer(httpPortWithAuth, withBasicAuth, false)
+	startServer(httpsPort, noBasicAuth, true)
+	startServer(httpsPortWithAuth, withBasicAuth, true)
 
-	wg.Add(1)
-	startServer(httpPort, noBasicAuth, false, wg)
-
-	wg.Add(1)
-	startServer(httpPortWithAuth, withBasicAuth, false, wg)
-
-	wg.Add(1)
-	startServer(httpsPort, noBasicAuth, true, wg)
-
-	wg.Add(1)
-	startServer(httpsPortWithAuth, withBasicAuth, true, wg)
-
-	wg.Wait()
+	select {}
```

Comment:
This doesn't feel semantically the same as the wait groups. Am I missing something? Also I think the log.Fatal in startServer is probably a mistake on my part.

Reply:
Both things block forever until one of the `startServer` exits the process with `log.Fatal`. The calls to `wg.Done()` are unreachable, so this wg was overkill.

I can rework this to not use log.Fatal and go back to workgroups if you prefer. In that case they wouldn't be overkill.